Actions/TelemetryHelper.psm1

```diff
@@ -212,4 +239,35 @@ function Add-TelemetryProperty() {
     }
 }
 
-Export-ModuleMember -Function Trace-Information, Trace-Exception, Add-TelemetryProperty
+<#
+    .SYNOPSIS
+    Writes a deprecation warning to telemetry and as a warning in the GitHub action
+
+    .DESCRIPTION
+    Writes a deprecation warning to telemetry and as a warning in the GitHub action
+
+    .PARAMETER Message
+    The message to log to telemetry
+
+    .PARAMETER DeprecationTag
+    The tag to use to link to the deprecation documentation
+
+    .EXAMPLE
+    Trace-DeprecationWarning -Message "This setting is deprecated. Use the new setting instead." -DeprecationTag 'SettingName'
+#>
+function Trace-DeprecationWarning {
+    param(
+        [Parameter(Mandatory = $true)]
+        [String] $Message,
+        [Parameter(Mandatory = $true)]
+        [String] $DeprecationTag
+    )
+
+    # Show deprecation warning in GitHub
+    OutputWarning -message "$Message. This warning will become an error in future versions. See https://aka.ms/ALGoDeprecations#$($DeprecationTag) for more information."
```

Comment:
Did we agree it'll be an error in the future? I thought AL-Go will quietly stop using this functionality. @freddydk what do you think?

Reply:
I added based on your suggestion 😆 I can remove the sentence from the message. That information should be present in the DEPRECATIONS.md anyway. 

As for what should happen after April 1st, we might want to consider throwing an error. Quietly removing it might lead to unexpected behavior if a partner misses the deprecation message. 

That will not be part of this PR though.

Reply:
Agree with not having this added to the message here - some deprecations might become errors, others might just stop working.